### PR TITLE
Update purldb doc

### PR DESCRIPTION
**Pull request by JonoYang**
This PR documents the `Filter by checksum` action on Packages and and `index_packages` on Resources api endpoints.

### CODE REVIEW

File: docs/source/purldb/rest_api.rst [hunk 1]
```diff
@@ -834,6 +866,87 @@ https://scancodeio.readthedocs.io/en/latest/built-in-pipelines.html.
...
+---------------
+
+index_packages
+^^^^^^^^^^^^^^
+
+Take a list of ``packages`` (where each item is a dictionary containing either PURL
+or versionless PURL along with vers range, optionally with source package PURL)
+and index it.
+Also each package can have list of ``addon_pipelines`` to run on the package.
+Find all addon pipelines [here.](https://scancodeio.readthedocs.io/en/latest/built-in-pipelines.html)
```
Comment (AyanSinhaMahapatra) [inline docs/source/purldb/rest_api.rst:879]:
```suggestion
Find all addon pipelines `here. <https://scancodeio.readthedocs.io/en/latest/built-in-pipelines.html>`_
```
This needs to be in RST syntax :P

File: docs/source/purldb/rest_api.rst [hunk 2]
```diff
@@ -834,6 +866,87 @@ https://scancodeio.readthedocs.io/en/latest/built-in-pipelines.html.
...
+
+
+If ``reindex`` flag is True then existing package will be rescanned, if ``reindex_set``
+is True then all the package in the same set will be rescanned.
+If reindex flag is set to true then all the non existing package will be indexed.
+
+**Note:** When a versionless PURL is supplied without a vers range, then all the versions
```
Comment (AyanSinhaMahapatra) [inline docs/source/purldb/rest_api.rst:886]: You can have this in a ``note`` directive like this:
https://github.com/nexB/scancode-toolkit/blob/develop/docs/source/getting-started/install.rst?plain=1#L135

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
@JonoYang thanks!
See comments for a couple nits.

**Comment by JonoYang** [comment]
@AyanSinhaMahapatra Thanks for the review, I've made the suggested changes

**Pull request merged**